src/app/shared/components/components/edit-product-modal/edit-product-modal.component.ts

```diff
@@ -0,0 +1,48 @@
+import { Component, Output, Input, EventEmitter, OnInit } from '@angular/core';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { ProductsService } from 'src/app/core/products.service';
+import { Product } from '../../../../product.interface';
+
+@Component({
+  selector: 'app-edit-product-modal',
+  templateUrl: './edit-product-modal.component.html',
+  styleUrls: ['./edit-product-modal.component.scss'],
+})
+export class EditProductModalComponent implements OnInit {
+  @Input() productToUpdate!: Product;
+
+  private numberRegEx = /^\s*(?=.*[1-9])\d*(?:\.\d{1,2})?\s*$/;
```

Comment:
Consider extracting reused parts of the code to a seperate file, for example `app/utils/validation.ts`